### PR TITLE
#57345 Bump the minimum supported version of PHP to 7.0.0

### PR DESCRIPTION
**Pull request by johnbillion**
Trac ticket: https://core.trac.wordpress.org/ticket/57345

## Changes

* Bumps minimum supported PHP version from 5.6.20 to 7.0.0
* Updates the testing on CI to remove PHP 5 and the corresponding "slow" test handling
* Updates linting and Composer config
* Removes random_compat
* Removes special case handling of PHP < 7 in a few tests

## Not changed

* Two uses of `$HTTP_RAW_POST_DATA`:
  1. https://github.com/johnbillion/wordpress-develop/blob/b6046636f015f6e1fac6b565f883fa9679b90a72/src/xmlrpc.php#L18-L28 - It's not entirely clear why this code is in place, it appears unused in core but may be used by plugins. Ticket to be opened to discuss.
  2. https://github.com/johnbillion/wordpress-develop/blob/b6046636f015f6e1fac6b565f883fa9679b90a72/src/wp-includes/rest-api/class-wp-rest-server.php#L1404-L1418 - This can be replaced with `return file_get_contents( 'php://input' )` however it breaks the `test_serve_request_json_params_are_unslashed()` test which is therefore useless when testing against PHP >= 7. Ticket to be opened to address.
* `'5.6'` used as the minimum PHP version in mock data for themes in tests, no need to change these and cause unnecessary churn.
* PHP 5 compat code that will be addressed in https://core.trac.wordpress.org/ticket/57067.

### CODE REVIEW

File: tests/phpunit/tests/functions/wpListUtil.php [hunk 1]
```diff
@@ -1025,10 +1021,6 @@ public function data_wp_list_util_sort_object_arrays() {
 	 * @param bool   $preserve_keys Optional. Whether to preserve keys. Default false.
 	 */
 	public function test_wp_list_util_sort_php_7_or_greater( $expected, $target_array, $orderby = array(), $order = 'ASC', $preserve_keys = false ) {
```
Comment (GaryJones) [inline tests/phpunit/tests/functions/wpListUtil.php]: Rename the test as well? And the corresponding data provider method below?

File: composer.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
 		"issues": "https://core.trac.wordpress.org/"
 	},
 	"require": {
-		"php": ">=5.6"
+		"php": ">=7.0"
```
Comment (spacedmonkey) [inline composer.json:13]: Worth doing a composer update?

**Review by spacedmonkey — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (johnbillion) [reply composer.json:13]: The composer.lock file isn't committed to the repo. What am I missing?

File: tests/phpunit/tests/functions/wpListUtil.php (hunk 1 above)
Comment (johnbillion) [reply tests/phpunit/tests/functions/wpListUtil.php]: Done, thanks 👍

**Review by mukeshpanchal27 — COMMENTED** [review]
Thanks, @johnbillion, for the heads up. It looks like the following two instance have a `requires_php` value that also needs to be updated to `7.0.0`.

https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-admin/includes/class-wp-site-health.php#L3065
https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-admin/includes/class-wp-site-health.php#L3075

**Comment by johnbillion** [comment]
> It looks like the following two instance have a `requires_php` value that also needs to be updated

I don't believe they do need updating because it's just mock data to ensure the health check works, but it doesn't hurt to change it so I'll get them updated anyway.

File: src/readme.html [hunk 3]
```diff
@@ -51,7 +51,7 @@ <h2>Migrating from other systems</h2>
 
 <h2>System Requirements</h2>
 <ul>
-	<li><a href="https://secure.php.net/">PHP</a> version <strong>5.6.20</strong> or greater.</li>
+	<li><a href="https://secure.php.net/">PHP</a> version <strong>7.0</strong> or greater.</li>
```
Comment (audrasjb) [inline src/readme.html:54]: Shouldn't we set `7.0.33` (released on 6 Dec 2018) as a minimum?

File: src/wp-includes/version.php [hunk 4]
```diff
@@ -37,7 +37,7 @@
  *
  * @global string $required_php_version
  */
-$required_php_version = '5.6.20';
+$required_php_version = '7.0.0';
```
Comment (audrasjb) [inline src/wp-includes/version.php:40]: Shouldn't we set `7.0.33` (released on 6 Dec 2018) as a minimum?

**Review by audrasjb — COMMENTED** [review]
Thanks for the PR! 
I left two nitpick proposals about the exact supported PHP version :)

File: src/wp-includes/version.php (hunk 4 above)
Comment (johnbillion) [reply src/wp-includes/version.php:40]: Thanks for the suggestion Jb. When the last minimum version bump was made (see [#46594](https://core.trac.wordpress.org/ticket/46594)) the minimum version was originally set to 5.6.30 but this got reduced to 5.6.20 after concerns around common tooling that uses older patch versions and unnecessarily excluding sites that are running older patch versions.

There's no major technical advantage to WordPress core using a later patch version in the 7.0 branch as the minimum, the only one mentioned on [#57345](https://core.trac.wordpress.org/ticket/57345) so far is 7.0.10 for the `IMG_WEBP` constant which is a minor advantage. We're going to go with 7.0.0.

**Review by spacedmonkey — APPROVED** [review]
I approve this with pleasure.

**Comment by johnbillion** [comment]
https://core.trac.wordpress.org/changeset/56141

**Pull request closed without merging**